src/coreclr/jit/emit.h

```diff
@@ -446,6 +446,41 @@ enum idAddrUnionTag
     iaut_SHIFT = 2
 };
 
+enum EmitCallType
+{
+    EC_FUNC_TOKEN, //   Direct call to a helper/static/nonvirtual/global method (call addr with RIP-relative encoding)
+    EC_FUNC_TOKEN_INDIR, // Indirect call to a helper/static/nonvirtual/global method (call [addr]/call [rip+addr])
```

Comment:
Not all of these were defined for all platforms. e.g., arm64 only defined `TOKEN` and `INDIR_R`. Should we ifdef out the cases that aren't available on a platform to avoid mistakenly using them?

Reply:
Added xarch ifdefs for `EC_FUNC_TOKEN_INDIR` and `EC_FUNC_TOKEN_ARD`